ts/main.ts

```diff
@@ -134,46 +209,238 @@ const $sortByRatedOrViewed = document.getElementById(
     }
-    $myListView.replaceChildren($ulCount);
+    $addReviewForm.dataset.index = index;
+    $addReviewForm.dataset.tag = tag;
+  }
+});
+interface ReviewEntity {
+  id: number;
+  businessName: string;
+  businessTag: string;
+  businessIndex: string;
+  ratingValue: string;
+  titleValue: string;
+  messageValue: string;
+}
+
+const $myReviewsPage = document.querySelector(
+  'div[data-reviews-page="my-reviews-page"]',
+) as HTMLDivElement;
+if (!$myReviewsPage) {
+  throw new Error('$myReviewsPage query failed');
+}
+const $myReviews = document.querySelector(
+  'div[data-reviews-page="my-reviews-page"] .row',
+) as HTMLDivElement;
+if (!$myReviews) {
+  throw new Error('$myReviews query failed');
+}
+$addReviewForm.addEventListener('submit', (event: Event) => {
+  event.preventDefault();
+  const $formElements = $addReviewForm.elements as FormElements;
+  const titleValue = $formElements.title.value;
+  const ratingValue = $formElements.rating.value;
+  const messageValue = $formElements.message.value;
+  const index = $addReviewForm.dataset.index as string;
+  const tag = $addReviewForm.dataset.tag as string;
+  const review: ReviewEntity = {
+    businessName:
+      tag === '1'
+        ? businessesRating[Number(index)].name
+        : businessesCount[Number(index)].name,
+    businessIndex: index,
+    businessTag: tag,
+    ratingValue,
+    titleValue,
+    messageValue,
+    id: data.nextEntityId,
+  };
+  data.reviews.unshift(review);
+  data.nextEntityId++;
+  const Reviews = data.reviews;
+  for (const review of Reviews) {
+    const $divReview = createReviewsDOMTree(review);
+    $myReviews.append($divReview);
   }
+  $businessProfile.style.display = 'none';
+  $landingPage.style.display = 'none';
+  $entitiesView.style.display = 'none';
+  $myReviewsPage.style.display = 'block';
+  $addReviewsPage.style.display = 'none';
+  $addReviewForm.reset();
 });
+
+function createReviewsDOMTree(entity: ReviewEntity): HTMLDivElement {
+  const $mainDiv = document.createElement('div') as HTMLDivElement;
+  $mainDiv.className = 'my-review column-full column-half';
+  const $divId = document.createElement('div') as HTMLDivElement;
+  $divId.textContent = `Review ID: ${entity.id.toString()}`;
+  const $divName = document.createElement('div') as HTMLDivElement;
+  $divName.textContent = `Business Name: ${entity.businessName}`;
+  const $divTitle = document.createElement('div') as HTMLDivElement;
+  $divTitle.textContent = `Title: ${entity.titleValue}`;
+  const $divRating = document.createElement('div') as HTMLDivElement;
+  $divRating.textContent = `Rating: ${entity.ratingValue}`;
+  const $divMessage = document.createElement('div') as HTMLDivElement;
+  $divMessage.textContent = `Message: ${entity.messageValue}`;
+  const $divbutton = document.createElement('div') as HTMLDivElement;
+  $divbutton.className = 'update-button';
+  const $button = document.createElement('button') as HTMLButtonElement;
```

Comment:
You shouldn't need to utilize type assertion on these variables. TypeScript should be able to infer the correct type based off of the argument in `createElement`